public/pages/workflow_detail/workflow_detail.test.tsx

```diff
@@ -0,0 +1,91 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import React from 'react';
+import { render } from '@testing-library/react';
+import { Provider } from 'react-redux';
+import {
+  RouteComponentProps,
+  Route,
+  Switch,
+  Router,
+  Redirect,
+} from 'react-router-dom';
+import { WorkflowDetail } from './workflow_detail';
+import { WorkflowDetailRouterProps } from '../../pages';
+import '@testing-library/jest-dom';
+import { mockStore, resizeObserverMock } from '../../../test/utils';
+import { createMemoryHistory } from 'history';
+import { WORKFLOW_TYPE } from '../../../common';
+
+jest.mock('../../services', () => {
+  const { mockCoreServices } = require('../../../test');
+  return {
+    ...jest.requireActual('../../services'),
+    ...mockCoreServices,
+  };
+});
+
+const history = createMemoryHistory();
+window.ResizeObserver = resizeObserverMock;
+
+const renderWithRouter = (
+  workflowId: string,
+  workflowName: string,
+  workflowType: WORKFLOW_TYPE
+) => ({
+  ...render(
+    <Provider store={mockStore(workflowId, workflowName, workflowType)}>
+      <Router history={history}>
+        <Switch>
+          <Route
+            path="/workflow/:workflowId"
+            render={(props: RouteComponentProps<WorkflowDetailRouterProps>) => {
+              return <WorkflowDetail setActionMenu={jest.fn()} {...props} />;
+            }}
+          />
+          <Redirect from="/" to={`/workflow/${workflowId}`} />
```

Comment:
wondering why we need this redirect? Can't we specify the path directly on line 44?

Reply:
- @ohltyler, Removed **Redirect** and instead adding **initialEntries** within the history. 
- And also below path should not be removed. Whenever URL matches the below pattern then workflowId will be dynamically obtained from URL. Similar to the actual use case of workflowDetail Component. 
`path="/workflow/:workflowId"`